include/hipCPU/hip/detail/queue.hpp

```diff
@@ -103,7 +103,12 @@ async_queue::~async_queue()
 {
   halt();
 
-  assert(_enqueued_operations.empty());
+  #if !defined(NDEBUG)
+  {
+    std::unique_lock<std::mutex> lock(_mutex);
+    assert(_enqueued_operations.empty());
```

Comment:
Can you elaborate why this change is necessary? Because of `halt()`, all asynchronous operations should have finished and the threads rejoined by the time of the assertion. Am I missing something?

Reply:
Let me investigate this a bit more in depth, your point about joined threads ringed a bell...

Reply:
You were totally right, that synced scope was totally useless, TSAN is still pretty happy without it. Just dropped the commit.